python/lsst/db/utils.py

```diff
@@ -397,46 +397,52 @@ def loadSqlScript(conn, script, dbName=None):
                        database used by connection/engine.
     """
 
-    # check file, if it has 'read' attribute assume it's a file object
-    if not hasattr(script, 'read'):
-        script = open(script)
-
     url = conn.engine.url
     if url.get_backend_name() == "mysql":
 
+        # check file, if it has 'read' attribute assume it's a file object
+        cleanup = None
+        if not hasattr(script, 'read'):
+            script = open(script)
+            cleanup = script.close
+
         # make tmpfile to store credentials and options
         fd, fname = tempfile.mkstemp(text=True)
```

Comment:
Because it contains text? This should only matter for newline chars, I guess, not encoding. `mkstemp()` return OS-level file descriptor which cannot be used with `print`.